distutils/command/install.py

```diff
@@ -84,6 +84,22 @@
 SCHEME_KEYS = ('purelib', 'platlib', 'headers', 'scripts', 'data')
 
 
+def _load_schemes():
+    """
+    Extend default schemes with schemes from sysconfig.
+    """
+
+    schemes = dict(INSTALL_SCHEMES)
+
+    try:
+        import sysconfig
+        schemes.update(sysconfig.INSTALL_SCHEMES)
```

Comment:
Uh, we never made `_INSTALL_SCHEMES` public IIRC, so the attribute error you are getting should be because the `INSTALL_SCHEMES` never exists.

I think the only way to get the install schemes is by calling `get_paths` for each scheme in `get_scheme_names()`.

Reply:
(on phone so sorry if this is misformated)

```python
schemes.update({
    scheme: sysconfig.get_paths(scheme)
    for scheme in sysconfig.get_scheme_names()
})
```

Reply:
You also need to pass `expand=False` to `get_paths` to make sure the templated names are not substituted in.

Reply:
Oh, right, I was forgetting about that. But the scheme trmplating is different IIRC, so it may be easier to inject the schemes at evaluation time, so that we can have sysconfig do the expansition itself.
I am not sure what would be easier to implement, convert the templating or have sysconfig do the evaluation. @jaraco if you want I can look at the code later.

Reply:
My instinct is it would be preferable for sysconfig to do the expansion except for:

- Initially, distutils is going to be responsible for providing a fallback for older Pythons, so whatever is implemented in sysconfig will also need to have a fallback implementation here.
- There may be cases where distutils wishes to provide functionality beyond what sysconfig can offer. Take for example #58, where `implementation` and `implementation_lower` were added to the schemes to allow PyPy to have custom install schemes based on standard install schemes. Perhaps that won't be necessary, but I'd be cautious about blocking the possibility of distutils providing more complex installation scenarios than sysconfig allows.

For that reason, I'm going to recommend not to rely on sysconfig to do the expansion (yet), but let's consider it for a possible future enhancement.

Reply:
Fixed in 1d7c74836b.